### PR TITLE
feat(design-system): add Toggle component and stories

### PR DESCRIPTION
**Pull request by mantariksh**
Implements Toggle component.

### Inclusion of icon in thumb
The original Chakra UI Switch [implementation](https://github.com/chakra-ui/chakra-ui/blob/main/packages/switch/src/switch.tsx) did not have the option to include an icon within the thumb (circular part of the Switch). Since we need to have icons in the thumb, I copied the implementation and adapted it to include the correct icon (checked/unchecked/disabled).

### CODE REVIEW

File: frontend/src/theme/components/Toggle.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    },
+    track: {
+      bg: 'neutral.400',
+      _checked: {
+        bg: 'success.500',
+      },
+      p: 0,
+      _focus: {
+        // disable Chakra UI's default shadow
+        boxShadow: `none`,
```
Comment (karrui) [inline frontend/src/theme/components/Toggle.ts:48]: Could we not have used a double box shadow?

```
boxShadow: `0 0 0 2px white, 0 0 0 3px var(--chakra-colors-primary-500)`;
```

**Comment by mantariksh** [comment]
need to fix a11y, label is not linked to toggle

**Comment by mantariksh** [comment]
> need to fix a11y, label is not linked to toggle

done

File: frontend/src/theme/components/Toggle.ts (hunk 1 above)
Comment (mantariksh) [reply frontend/src/theme/components/Toggle.ts:48]: excuse me while I go drown myself

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**